bank.py

```diff
@@ -86,11 +96,17 @@ def change_user_pin(self, user: User, new_pin: str) -> None:
             return
 
         user.pin = user.encrypt_pin(new_pin)
-        self.save_users()
-
-    def save_users(self) -> None:
-        """loads all information about users in JSON"""
-        json_string = json.dumps([user.__dict__ for user in self.users])
-
-        with open("database.json", "w") as file:
-            file.write(json_string)
+        self.save_users(user)
+
+    def save_users(self, user: User) -> None:
```

Comment:
I don't really know if it's OK connecting to database several times. Maybe I must to do it only once?

Reply:
you need to connect each time when operating with it
if you will keep connection open is bad for performance